app/src/main/java/com/ncnf/home/ui/HomeFragment.java

```diff
@@ -61,6 +73,17 @@ public void goToFriends(View view){
         goToActivity(FriendsActivity.class);
     }
 
+    public void goToFindFriends(View view){
+        Intent intent;
+
+        if(!isConnected()){
+            intent = new Intent(getContext(), LoginActivity.class);
+            startActivity(intent);
+        } else {
+            gpsIsEnabled();
```

Comment:
I find the logic of this function a bit odd. From what I understand to launch the activity we call this function with a clear name, but then go from function to function until all the checks have passed and only three functions further does the activity actually launch? Wouldn't it be simpler to have this gpsIsEnabled be a function returning a boolean, getLocationPermission as well and if those two return true, then in the function goToFindFriends, launch the Friends Tracking activity?

Reply:
It's done this way to ensure that we know that GPS is enabled before asking for location permissions for this specific app. I can try to refactor it, but I don't know how android will react if a user enables location services on a specific app without having enabled it on the phone in general.